uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/API/Handler/GenericLeaveFieldPermissionsTest.php

```diff
@@ -0,0 +1,320 @@
+            'to_date' => '2017-02-01 00:00:00',
+            'to_date_type' => '1',
+            'sickness_reason' => '2',
+            'request_type' => 'sickness',
+            'is_deleted' => '0',
+            'from_date_amount' => '0.00',
+            'to_date_amount' => '0.00',
+            'balance_change' => -1,
+            'dates' =>
+              [
+                0 =>
+                  [
+                    'id' => '63',
+                    'date' => '2017-02-01',
+                  ],
+              ],
+          ],
+        5 =>
+          [
+            'id' => '28',
+            'type_id' => '3',
+            'contact_id' => '209',
+            'status_id' => '6',
+            'from_date' => '2017-02-01 00:00:00',
+            'from_date_type' => '1',
+            'to_date' => '2017-02-01 00:00:00',
+            'to_date_type' => '1',
+            'sickness_reason' => '2',
+            'request_type' => 'sickness',
+            'is_deleted' => '0',
+            'from_date_amount' => '0.00',
+            'to_date_amount' => '0.00',
+            'balance_change' => -1,
+            'dates' =>
+              [
+                0 =>
+                  [
+                    'id' => '63',
+                    'date' => '2017-02-01',
+                  ],
+              ],
+          ],
+      ],
+  ];
+
+  public function setUp() {
+
+  }
+
+  public function testProcessWhenUserIsStaff() {
+    //User is Staff with ID of 204 and has full access to only his data
+    $this->setPermissions();
+    $leaveRequestRights = $this->prophesize(LeaveRequestRightsService::class);
+    $leaveRequestRights->getLeaveContactsCurrentUserHasAccessTo()->willReturn([204]);
+    $apiRequest = [];
+    $genericFieldHandler = new GenericLeaveFieldPermissions($apiRequest, $leaveRequestRights->reveal());
+
+    $results = $this->sampleData;
+    $expectedParams = $this->sampleData;
+    $expectedParams['values'][0]['from_date_amount'] = '';
+    $expectedParams['values'][0]['to_date_amount'] = '';
+    $expectedParams['values'][0]['balance_change'] = '';
+    $expectedParams['values'][0]['type_id'] = '';
+    $expectedParams['values'][3]['from_date_amount'] = '';
+    $expectedParams['values'][3]['to_date_amount'] = '';
+    $expectedParams['values'][3]['balance_change'] = '';
+    $expectedParams['values'][3]['toil_duration'] = '';
+    $expectedParams['values'][3]['toil_to_accrue'] = '';
+    $expectedParams['values'][3]['toil_expiry_date'] = '';
+    $expectedParams['values'][3]['type_id'] = '';
+    $expectedParams['values'][5]['sickness_reason'] = '';
+    $expectedParams['values'][5]['from_date_amount'] = '';
+    $expectedParams['values'][5]['to_date_amount'] = '';
+    $expectedParams['values'][5]['balance_change'] = '';
+    $expectedParams['values'][5]['type_id'] = '';
+    $genericFieldHandler->process($results);
+
+    $this->assertEquals($expectedParams, $results);
+  }
+
+  public function testProcessWhenUserIsManager() {
```

Comment:
The only reason we have this test is because we allow different restricted values for managers and staff members. But do we really need that? After reading the code, I'm not so sure. Compare this test with the staff one for example. They are pretty much the same. The only difference is that you need to set the permission first, but the actual logic works based on what `getLeaveContactsCurrentUserHasAccessTo()` returns and the permission is not used there.

Also there's the problem that this is changing the concept of what is a "Leave Approver". Before a "Leave Approver" was anyone with a Leave Approver relationship with someone else. That it is that + someone who has the "manager leave request in SSP" permission. If we don't differentiate the restricted values for managers and staff members, we would be able to keep the old concept and not have to rely on the permission.